gax/src/main/java/com/google/api/gax/tracing/TracedServerStreamingCallable.java

```diff
@@ -0,0 +1,138 @@
+@InternalApi("For internal use by google-cloud-java clients only")
+public class TracedServerStreamingCallable<RequestT, ResponseT>
+    extends ServerStreamingCallable<RequestT, ResponseT> {
+
+  @Nonnull private final ApiTracerFactory tracerFactory;
+  @Nonnull private final SpanName spanName;
+  @Nonnull private final ServerStreamingCallable<RequestT, ResponseT> innerCallable;
+
+  public TracedServerStreamingCallable(
+      @Nonnull ServerStreamingCallable<RequestT, ResponseT> innerCallable,
+      @Nonnull ApiTracerFactory tracerFactory,
+      @Nonnull SpanName spanName) {
+    this.tracerFactory = Preconditions.checkNotNull(tracerFactory, "tracerFactory can't be null");
+    this.spanName = Preconditions.checkNotNull(spanName, "spanName can't be null");
+    this.innerCallable = Preconditions.checkNotNull(innerCallable, "innerCallable can't be null");
+  }
+
+  @Override
+  public void call(
+      RequestT request, ResponseObserver<ResponseT> responseObserver, ApiCallContext context) {
+
+    ApiTracer tracer = tracerFactory.newTracer(spanName);
+    TracedResponseObserver<ResponseT> tracedObserver =
+        new TracedResponseObserver<>(tracer, responseObserver);
+
+    context = context.withTracer(tracer);
+
+    try {
+      innerCallable.call(request, tracedObserver, context);
+    } catch (RuntimeException e) {
+      tracedObserver.onError(e);
+      throw e;
+    }
+  }
+
+  /** A {@link ResponseObserver} to mark a started operation trace as finished. */
+  private static class TracedResponseObserver<ResponseT> implements ResponseObserver<ResponseT> {
+    private final ApiTracer tracer;
+    private final ResponseObserver<ResponseT> innerObserver;
+    private volatile boolean wasCancelled;
+
+    private TracedResponseObserver(
+        @Nonnull ApiTracer tracer, @Nonnull ResponseObserver<ResponseT> innerObserver) {
+      this.tracer = Preconditions.checkNotNull(tracer, "tracer can't be null");
+      this.innerObserver = Preconditions.checkNotNull(innerObserver, "innerObserver can't be null");
+    }
+
+    @Override
+    public void onStart(final StreamController controller) {
+      innerObserver.onStart(
+          new StreamController() {
+            @Override
+            public void cancel() {
+              wasCancelled = true;
+              controller.cancel();
+            }
+
+            @Override
+            public void disableAutoInboundFlowControl() {
+              controller.disableAutoInboundFlowControl();
+            }
+
+            @Override
+            public void request(int count) {
+              controller.request(count);
+            }
+          });
+    }
+
+    @Override
+    public void onResponse(ResponseT response) {
+      tracer.responseReceived();
+      innerObserver.onResponse(response);
+    }
+
+    @Override
+    public void onError(Throwable t) {
+      if (wasCancelled) {
+        tracer.operationCancelled();
```

Comment:
From [bidi PR](https://github.com/googleapis/gax-java/pull/636/files#r251552448)
What would be the value of Throwable t here? Can it really be dropped? Won't it be same as cancellationCause? If no, why?

Reply:
This is to workaround the the retry implementation using cancellations to end polling. I don't want these to be traced as cancellations. I added a comment explaining the motivation